test/unit/storage/FileResourceStore.test.ts

```diff
@@ -0,0 +1,572 @@
+    fs.createWriteStream = jest.fn();
+    const writeStream = {
+      on(name: string, func: () => void): any {
+        if (name === 'finish') {
+          func();
+        }
+        return this;
+      },
+      once(): any {
+        return this;
+      },
+      emit(): any {
+        return true;
+      },
+      write(): any {
+        return true;
+      },
+      end(): any {
+        // Empty
+      },
+    };
+    (fs.createWriteStream as jest.Mock).mockReturnValue(writeStream);
+
+    readableMock = {
+      on(name: string, func: () => void): any {
+        if (name === 'finish') {
+          func();
+        }
+        return this;
+      },
+      pipe(): any {
+        return this;
+      },
+    } as unknown as Readable;
+  });
+
+  it('errors if a resource was not found.', async(): Promise<void> => {
+    (fsPromises.lstat as jest.Mock).mockImplementationOnce((): any => {
+      throw new Error('Path does not exist.');
+    });
+    (fsPromises.lstat as jest.Mock).mockImplementationOnce((): any => {
+      throw new Error('Path does not exist.');
+    });
+    await expect(store.getRepresentation({ path: 'http://wrong.com/wrong' })).rejects.toThrow(NotFoundHttpError);
+    await expect(store.getRepresentation({ path: `${base}wrong` })).rejects.toThrow(NotFoundHttpError);
+    await expect(store.addResource({ path: 'http://wrong.com/wrong' }, representation))
+      .rejects.toThrow(NotFoundHttpError);
+    await expect(store.deleteResource({ path: 'wrong' })).rejects.toThrow(NotFoundHttpError);
+    await expect(store.deleteResource({ path: `${base}wrong` })).rejects.toThrow(NotFoundHttpError);
+    await expect(store.setRepresentation({ path: 'http://wrong.com/' }, representation))
+      .rejects.toThrow(NotFoundHttpError);
+  });
+
+  it('errors when modifying resources.', async(): Promise<void> => {
+    await expect(store.modifyResource()).rejects.toThrow(Error);
+  });
+
+  it('errors for wrong input data types.', async(): Promise<void> => {
+    (representation as any).dataType = DATA_TYPE_QUAD;
+    await expect(store.addResource({ path: base }, representation)).rejects.toThrow(UnsupportedMediaTypeHttpError);
+    await expect(store.setRepresentation({ path: `${base}foo` }, representation)).rejects
+      .toThrow(UnsupportedMediaTypeHttpError);
+  });
+
+  it('can write and read a container.', async(): Promise<void> => {
+    // Mock the fs functions.
+    // Add
+    (fsPromises.mkdir as jest.Mock).mockReturnValueOnce(true);
+
+    // Mock: Get
+    stats.isFile = (): any => false;
+    stats.isDirectory = (): any => true;
+    (fsPromises.lstat as jest.Mock).mockReturnValueOnce(stats);
+    (fsPromises.readdir as jest.Mock).mockReturnValueOnce([]);
+    (fs.createReadStream as jest.Mock).mockImplementationOnce((): any => new Error('Metadata file does not exist.'));
+
+    // Write container (POST)
+    (representation as any).metadata = { linkRel: { type: new Set() }, slug: 'myContainer/', raw: []};
```

Comment:
Otherwise I get `TS2741: Property 'profiles' is missing in type '{ linkRel: { type: Set<string>; }; slug: string; raw: never[]; }' but required in type 'RepresentationMetadata'.`
I could also add `profiles: []` everywhere if you prefer?

Reply:
Actually you can set `profiles` to be an optional parameter in the metadata. This is something that needs to change but hasn't been done yet.